runtime/integration-tests/src/generic/cases/proxy.rs

```diff
@@ -44,14 +56,68 @@ fn configure_proxy_and_transfer<T: Runtime>(proxy_type: T::ProxyType) -> Dispatc
 	configure_proxy_and_call::<T>(env, proxy_type, call)
 }
 
-fn configure_proxy_and_x_transfer<T: Runtime>(proxy_type: T::ProxyType) -> DispatchResult {
-	/*
-	let env = // TODO: initialize fudge env with xcm support
-	let call = // TODO: create a 'pallet_restricted_xtokens::transfer' call
+fn configure_proxy_and_x_transfer<T: Runtime + FudgeSupport>(
+	proxy_type: T::ProxyType,
+) -> DispatchResult {
+	let mut env = FudgeEnv::<T>::from_parachain_storage(
+		Genesis::default()
+			.add(genesis::balances::<T>(
+				T::ExistentialDeposit::get() + FOR_FEES,
+			))
+			.add(genesis::tokens(vec![(Usd6::ID, Usd6::ED)]))
+			.storage(),
+	);
+
+	setup_xcm(&mut env);
+
+	env.parachain_state_mut(|| {
+		register_currency::<T, Usd6>();
+	});
+
+	let call = pallet_restricted_xtokens::Call::transfer {
+		currency_id: Usd6::ID,
+		amount: TRANSFER_AMOUNT,
+		dest: Box::new(
+			MultiLocation::new(
+				1,
+				X2(
+					Parachain(T::FudgeHandle::SIBLING_ID),
+					Junction::AccountId32 {
+						id: TO.into(),
+						network: None,
+					},
+				),
+			)
+			.into(),
+		),
+		dest_weight_limit: WeightLimit::Unlimited,
+	}
+	.into();
+
+	configure_proxy_and_call::<T>(env, proxy_type, call)
+}
+
+pub fn register_currency<T: Runtime + FudgeSupport, C: CurrencyInfo>() {
+	let meta: AssetMetadata<Balance, CustomMetadata> = AssetMetadata {
+		decimals: C::DECIMALS,
+		name: C::NAME.into(),
+		symbol: C::SYMBOL.into(),
+		existential_deposit: C::ED,
+		location: Some(VersionedMultiLocation::V3(MultiLocation::new(
+			1,
+			X1(Parachain(T::FudgeHandle::SIBLING_ID)),
+		))),
+		additional: CustomMetadata {
+			transferability: CrossChainTransferability::Xcm(Default::default()),
+			..CustomMetadata::default()
+		},
+	};
 
-	configure_proxy_and_call::<T>(env, proxy_type, call);
-	*/
-	todo!()
+	assert_ok!(orml_asset_registry::Pallet::<T>::register_asset(
+		<T as frame_system::Config>::RuntimeOrigin::root(),
+		meta,
+		Some(C::ID)
+	));
```

Comment:
Question:

Why do we need to register the currency in the `asset_registry`? is it a requirement for `xtokens`?

Reply:
It's a requirement for `orml_xtokens` since it will eventually check if the currency that we are trying to transfer is registered.